tests/components/blebox/test_cover.py

```diff
@@ -0,0 +1,381 @@
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My gatebox"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
+    assert device.model == "gateBox"
+    assert device.sw_version == "1.23"
+
+
+@pytest.mark.parametrize("wrapper", ALL_COVER_FIXTURES, indirect=["wrapper"])
+async def test_open(wrapper, hass, config):
+    """Test cover opening."""
+
+    feature_mock, entity_id = wrapper
+
+    def initial_update():
+        feature_mock.state = 3  # manually stopped
+
+    def open_gate():
+        feature_mock.state = 1  # opening
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    feature_mock.async_open = AsyncMock(side_effect=open_gate)
+
+    await async_setup_entity(hass, config, entity_id)
+    assert hass.states.get(entity_id).state == STATE_CLOSED
+
+    feature_mock.async_update = AsyncMock()
+    await hass.services.async_call(
+        "cover", SERVICE_OPEN_COVER, {"entity_id": entity_id}, blocking=True,
+    )
+    assert hass.states.get(entity_id).state == STATE_OPENING
+
+
+@pytest.mark.parametrize("wrapper", ALL_COVER_FIXTURES, indirect=["wrapper"])
+async def test_close(wrapper, hass, config):
+    """Test cover closing."""
+
+    feature_mock, entity_id = wrapper
+
+    def initial_update():
+        feature_mock.state = 4  # open
+
+    def close():
+        feature_mock.state = 0  # closing
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    feature_mock.async_close = AsyncMock(side_effect=close)
+
+    await async_setup_entity(hass, config, entity_id)
+    assert hass.states.get(entity_id).state == STATE_OPEN
+
+    feature_mock.async_update = AsyncMock()
+    await hass.services.async_call(
+        "cover", SERVICE_CLOSE_COVER, {"entity_id": entity_id}, blocking=True
+    )
+    assert hass.states.get(entity_id).state == STATE_CLOSING
+
+
+def opening_to_stop_feature_mock(feature_mock):
+    """Return an mocked feature which can be updated and stopped."""
+
+    def initial_update():
+        feature_mock.state = 1  # opening
+
+    def stop():
+        feature_mock.state = 2  # manually stopped
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    feature_mock.async_stop = AsyncMock(side_effect=stop)
+
+
+@pytest.mark.parametrize("wrapper", FIXTURES_SUPPORTING_STOP, indirect=["wrapper"])
+async def test_stop(wrapper, hass, config):
+    """Test cover stopping."""
+
+    feature_mock, entity_id = wrapper
+    opening_to_stop_feature_mock(feature_mock)
+
+    await async_setup_entity(hass, config, entity_id)
+    assert hass.states.get(entity_id).state == STATE_OPENING
+
+    feature_mock.async_update = AsyncMock()
+    await hass.services.async_call(
+        "cover", SERVICE_STOP_COVER, {"entity_id": entity_id}, blocking=True
+    )
+    assert hass.states.get(entity_id).state == STATE_OPEN
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)